pkg/cmd/repo_list.go

```diff
@@ -39,16 +38,19 @@ func newRepoListCmd(out io.Writer) *cobra.Command {
 		ValidArgsFunction: noMoreArgsCompFunc,
 		RunE: func(_ *cobra.Command, _ []string) error {
 			f, _ := repo.LoadFile(settings.RepositoryConfig)
-			if len(f.Repositories) == 0 && !(outfmt == output.JSON || outfmt == output.YAML) {
-				return errors.New("no repositories to show")
+			if len(f.Repositories) == 0 {
+				switch outfmt {
+				case output.Table:
+					_, _ = fmt.Fprintln(out, "no repositories to show")
+					return nil
+				case output.JSON, output.YAML:
+					return outfmt.Write(out, &repoListWriter{repos: []*repo.Entry{}})
+				}
```

Comment:
The empty repositories + JSON or YAML case is already handled by line 50. There is no need to handle it specially here I think. I think all that is needed for this PR is:

```suggestion
			if len(f.Repositories) == 0 && !(outfmt == output.JSON || outfmt == output.YAML) {
				_, _ = fmt.Fprintln(out, "no repositories to show")
					return nil
```